### ISSUE TITLE
GPy's CI is currently broken

### ISSUE THREAD

**Issue by ekalosak** [SheffieldML/GPy#905]
# Summary

The builds on `osx` time out when trying to download miniconda using the [download_miniconda.sh](https://github.com/mzwiessele/travis_scripts/blob/master/download_miniconda.sh) script.

# Evidence

In #904 , having an inert change that triggers the TravisCI integration tests, the builds on Mac OsX fail by timing out when installing pandoc via `brew install pandoc`. This brew command is the difference between the linux and mac builds (as defined in the donwload_miniconda.sh script) that causes this timeout behavior. The logs are in TravisCI, and [this run](https://travis-ci.org/github/SheffieldML/GPy/jobs/767185966) provides an example of the issue.

# Things I've tried
1. Thinking that this might be a problem with an upstream service (e.g. with brew) that will clear up if we wait, I reinstalled pandoc on my machine to probe those upstream services. They all work fine.
2. Taking the advice at the bottom of the logs indicating that the build failed due to a timeout, I introduced the `travis_wait` command to the CI config that doubles (by default) the timeout delay. This did not fix the issue, so I'm led to believe that some lag is not the issue.
3. From the TravisCI logs, the build process complains about another problem, this time with the outdated osx and xcode versions used in `.travis.yaml`, so I've pinned an updated osx version in the CI config. That build is in progress [here](https://travis-ci.org/github/SheffieldML/GPy/builds/767187853).

# Bottom line
The current CI is broken. I know this because I tested the CI on an inert change to the README.

**Comment by ekalosak** [comment]
@zhenwendai @lawrennd @mzwiessele Appears that the macOS version update introduced in #904 fixes the CI. Looks like it boiled down to `brew` stopping support for the implicit Mac OsX version that TravisCI uses. Updating the OsX version was all that was required to fix it.
